### ISSUE TITLE
DataTable: RowToggle collapse does not skip children

### ISSUE THREAD

**Issue by DaanVanYperen** [primefaces/primefaces#6596]
**Describe the defect**
RowToggle ajax event with `skipChildren="true"` only skips children on expand, not collapse.

**Reproducer**
https://github.com/DaanVanYperen/primefaces-test-skipchildren-missing-from-rowToggle-collapse/

**Environment:**
 - PF Version: _8.0.5_
 - JSF + version: reproducer defaults.

**To Reproduce**
1. Expand datatable row.
2. Check posted form data, it contains `skipChildren`
![image](https://user-images.githubusercontent.com/6305602/101001497-7f5e8c00-355f-11eb-8a89-c656bf46d8b9.png)
3. Collapse datatable row.
4. Check posted form data, `skipChildren` is missing and children are processed.
![image](https://user-images.githubusercontent.com/6305602/101001657-addc6700-355f-11eb-9028-720f90507098.png)

**Expected behavior**
SkipChildren attribute on rowToggle ajax event also applies to collapse.

**Example XHTML**
```html
        <h:form id="frmTest">
            <p:dataTable value="#{testView.rows}" var="iRow" skipChildren="false">
                <p:ajax event="rowToggle" skipChildren="true" listener="#{testView.onRowToggle}" />
                <p:column>
                    <p:rowToggler />
                </p:column>
                <p:column>
                    #{iRow}
                </p:column>
                <p:column>
                    <h:inputText value="#{testView.input}" />
                </p:column>
                <p:rowExpansion>
                    Detail of #{iRow}
                </p:rowExpansion>
            </p:dataTable>
        </h:form>
```

**Example Bean**
```java
@Named
@ViewScoped
public class TestView implements Serializable {

    private List<String> rows;

    @PostConstruct
    public void init() {
        rows = Arrays.asList("Apple");
    }

    public List<String> getRows() {
        return rows;
    }

    public void onRowToggle(ToggleEvent event) {
        System.out.println("Row toggled");
    }

    public String getInput() {
        return "";
    }
    public void setInput(String value) {
        System.out.println("skipChildren ignored!");
    }
}
```
**Workaround**

This is the bandaid we applied that works for our usecase.

```java
    PrimeFaces.widget.DataTable = PrimeFaces.widget.DataTable.extend(
        {
            fireRowCollapseEvent: function(row) {
                var rowIndex = this.getRowMeta(row).index;

                if(this.hasBehavior('rowToggle')) {
                    var ext = {
                        params: [
                            {
                                name: this.id + '_collapsedRowIndex',
                                value: rowIndex
                            },
                            // workaround start
                            {   name: this.id + '_skipChildren',
                                value: true
                            }
                            // workaround end
                        ]
                    };

                    this.callBehavior('rowToggle', ext);
                }
            },
        }
    );
```

**Comment by melloware** [comment]
@DaanVanYperen can you add this JS to your application JS file that loads after PF and let me know if its fixed. It should be.
```javascript
if (PrimeFaces.widget.DataTable) {
    PrimeFaces.widget.DataTable.prototype.fireRowCollapseEvent = function(row) {
        var rowIndex = this.getRowMeta(row).index;

        if (this.hasBehavior('rowToggle')) {
            var ext = {
                params: [{
                        name: this.id + '_collapsedRowIndex',
                        value: rowIndex
                    },
                    {
                        name: this.id + '_encodeFeature',
                        value: true
                    },
                    {
                        name: this.id + '_skipChildren',
                        value: true
                    }
                ]
            };
            this.callBehavior('rowToggle', ext);
        }
    };
}
```

**Comment by melloware** [comment]
Ahh nevermind I see you already tested it.

**Comment by melloware** [comment]
Thank you for reporting!  Fixed for 9.0

**Comment by DaanVanYperen** [comment]
> Thank you for reporting! Fixed for 9.0

That was quick! Cheers.
